File: 88.merge-sorted-array.rb
#
# @lc app=leetcode id=88 lang=ruby
#
# [88] Merge Sorted Array
#

# @lc code=start
# @param {Integer[]} nums1
# @param {Integer} m
# @param {Integer[]} nums2
# @param {Integer} n
# @return {Void} Do not return anything, modify nums1 in-place instead.
def merge(nums1, m, nums2, n)
  n.times do |i|
    nums1[-1 + (-1) * i] = nums2[i]
  end
  nums1.sort!
end
# @lc code=end

File: 88.merge-sorted-array.py
#
# @lc app=leetcode id=88 lang=python3
#
# [88] Merge Sorted Array
#

# @lc code=start
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        for i in range(m + n):
            if len(nums2) == 0: continue
            if nums1[i] < nums2[0]: continue
            nums1.insert(i, nums2[0])
            nums2.pop(0)
        for i in range(n):
            nums1.pop(-1)
        for i in nums2:
            nums1.append(i)
# @lc code=end

